source/_docs/installation/synology.markdown

```diff
@@ -2,78 +2,287 @@
+</p>
 
+```bash
+# sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+```
+Refresh package lists for newly added Docker repository:
+```bash
+# sudo apt-get update
+```
+Install Docker:
+```bash
+# sudo apt-get install docker-ce
+```
+---
+### {% linkable_title Installing Docker on [Solus](https://getsol.us) %}
 
-Check the path to python3 (assumed to be /volume1/@appstore/py3k/usr/local/bin)
-
+Update package lists:
 ```bash
-# cd /volume1/@appstore/py3k/usr/local/bin
+# sudo eopkg update-repo
+```
+Install Docker:
+```bash
+# sudo eopkg install docker
+```
+Enable the docker service at system boot:
+```bash
+# sudo systemctl enable docker
+```
+Start the docker service:
+```bash
+# sudo systemctl start docker
+```
+---
+To enable your user account to manage Docker without superuser privileges, add your user account to the “docker” group:
+```bash
+# sudo gpasswd -a YOUR_USER_ACCOUNT docker
 ```
 
-Install PIP (Python's package management system)
+You will now need to get the [spksrc framework and Docker container](https://github.com/SynoCommunity/spksrc#docker) for the compiling environment.
 
+Install Git
+#### Ubuntu/Mint
+```bash
+# sudo apt-get install git
+```
+#### Solus
 ```bash
-# ./python3 -m ensurepip
+# sudo eopkg install git
 ```
+Fork and clone spksrc, this will make a directory named “spksrc”.
+```bash
+git clone https://You@github.com/You/spksrc.git ~/spksrc
+```
+Download the spksrc Docker container:
+```bash
+docker pull synocommunity/spksrc
+```
+
+Now you need to edit 2 files.
+You need to do this to get “cross compiled module package” files, which you require for Home Assistant to able to be installed.
+These edits also enable you to use the “[Cloud](/components/cloud/)” component and fix the OpenSSL errors when using the “[Xiaomi_Aqara](/components/Xiaomi_Aqara/)” component.
 
-Use PIP to install Homeassistant package 0.64.3
 
+Edit “*~/spksrc/spksrc/spk/python3/src/requirements.txt*”, add at the end of the file:
+```
+##Cross compilation requirements for installing Home Assistant (Tested to work on 83.3).
+##In the future, the requirements may change (need newer version to work, eg), modify as needed.
+cffi==1.11.5
+bcrypt==3.1.4
+cryptography==2.3.1
+#Cross compilation requirement for installing “Warrent” module (Needed by “Cloud” Component)
+pycryptodome==3.7.2
+```
+
+Edit “*~/spksrc/spksrc/spk/python3/src/Makefile*”, add above the line that says “<b>include ../../mk/spksrc.spk.mk</b>”:
+```makefile
+# Needed to fix “_openssl.so: undefined symbol: pthread_atfork” error caused by lack of libpthread linkage on Synology (Needed for SSL and “xiaomi_aqara” component)
+export CFLAGS=-pthread
+```
+
+## {% linkable_title Compiling the Python3 package %}
+Run the container, this will make your terminal run inside the Docker container:
 ```bash
-# ./python3 -m pip install homeassistant==0.64.3
+# docker run -it -v ~/spksrc:/spksrc synocommunity/spksrc /bin/bash
```

Comment:
Can a regular user run Docker?

Reply:
By regular user, do you mean a less technical user or a less privileged user?
If by less privileged user, the use of superuser is needed when installing packages and using Systemd to activate docker service, the rest has been written so it can be run as normal user.

The "docker run" command is done without superuser by following the guide and running:
-
To enable your user account to manage Docker without superuser privileges, add your user account to the “docker” group:
```bash
# sudo gpasswd -a YOUR_USER_ACCOUNT docker
```
-

If you mean less technical user, the guide has been made to be as simple and explaining as possible, as you could for the most part just copy paste commands and everything works, until the parts you need to edit files or add slightly modify commands to insert user or IP.
The use of Home Assistant already sets a point were you need some technical knowledge (in the sense of terminal, "scary" commands and the sort).

Reply:
I was talking about a less privileged system user.